src/core/core.config.js

```diff
@@ -31,18 +31,26 @@ function axisFromPosition(position) {
   }
 }
 
-export function determineAxis(id, scaleOptions) {
-  if (id === 'x' || id === 'y') {
+export function determineAxis(name, scaleOptions) {
+  let id = name[0].toLowerCase();
+
+  if (id === 'x' || id === 'y' || id === 'r') {
+    return id;
+  }
```

Comment:
```js
scales: {
  xavier: {
    position: 'left',
    axis: 'y'
  }
}
```

would return `x` (unexpectedly)

IMO the axis determination from first letter should still be the last resort.